### ISSUE TITLE
error:1E08010C:DECODER routines::unsupported when using Google Cloud Service Account

### ISSUE THREAD

**Issue by gadgrandez** [n8n-io/n8n#10347]
### Bug Description

I am unable to interact with any Google Cloud service that uses a service account. I have tried with BigQuery and Vertex AI so far. The error in the console shows the error: call to gauth.request message: error:1E08010C:DECODER routines::unsupported

### To Reproduce

1. Create a BigQuery Node.
2. Create or attach a Service Account to the BigQuery Node.
3. Insert a Project ID and a query in the respective fields.
4. Run the workflow with a "click" trigger.

### Expected behavior

The BigQuery Node should run successfully and return results.

### Operating System

Docker

### n8n Version

1.53.2

### Node.js Version

20

### Database

PostgreSQL

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @gadgrandez 

That looks like it can't decode the private key which normally means you may not have pasted in the private key value correctly.

From the json file you get from Google can you make sure you copy everything between the `"` for the `private_key` value.

**Comment by gadgrandez** [comment]
Hello @Joffcom. I have found the same instructions in various forums, however I have checked twice and copied it as is. Even when configuring the credential I receive the message "Connection tested successfully"
![Screenshot_3](https://github.com/user-attachments/assets/07a53ba9-c69e-4f71-8d87-696976813cea)

The same error occurs on various nodes that use service accounts. I don't know since when it happens, but I haven't had any problems for a few months.

**Comment by Joffcom** [comment]
Hey @gadgrandez 

That is interesting is it just those 2 nodes? I wonder if we don't fully support service accounts with them yet.

I will check on Monday morning.

**Comment by gadgrandez** [comment]
Ok, I've checked again. The error occurs on AI nodes using the "Google Vertex Chat Model".

Error: error:1E08010C:DECODER routines::unsupported at Sign.sign (node:internal/crypto/sig:128:29) at Object.sign (/usr/local/lib/node_modules/n8n/node_modules/jwa/index.js:152:45) at Object.jwsSign [as sign] (/usr/local/lib/node_modules/n8n/node_modules/jws/lib/sign-stream.js:32:24) at GoogleToken.requestToken (/usr/local/lib/node_modules/n8n/node_modules/gtoken/build/src/index.js:232:31) at GoogleToken.getTokenAsyncInner (/usr/local/lib/node_modules/n8n/node_modules/gtoken/build/src/index.js:166:21) at GoogleToken.getTokenAsync (/usr/local/lib/node_modules/n8n/node_modules/gtoken/build/src/index.js:145:55) at GoogleToken.getToken (/usr/local/lib/node_modules/n8n/node_modules/gtoken/build/src/index.js:97:21) at JWT.refreshTokenNoCache (/usr/local/lib/node_modules/n8n/node_modules/google-auth-library/build/src/auth/jwtclient.js:172:36) at JWT.refreshToken (/usr/local/lib/node_modules/n8n/node_modules/google-auth-library/build/src/auth/oauth2client.js:153:24) at JWT.getRequestMetadataAsync (/usr/local/lib/node_modules/n8n/node_modules/google-auth-library/build/src/auth/oauth2client.js:298:28)

![Screenshot_4](https://github.com/user-attachments/assets/7fd9003e-3661-4455-b6ac-74685742ea25)

**Comment by Joffcom** [comment]
Hey @gadgrandez,

Perfect so just to confirm this is not related to BigQuery and it is more of an issue with the Vertex AI node?

**Comment by Joffcom** [comment]
Internal reference for this one is `AI-276`

**Comment by gadgrandez** [comment]
Hi @Joffcom. Yes, the issue is with Vertex AI Node.

**Comment by tiburssio** [comment]
I'm facing same problem here on version 1.53.4

**Comment by anandarh** [comment]
Up, I also had the same problem on Vertex AI Node.

**Comment by Beedataco** [comment]
We have this error too!

**Comment by hen0liveira** [comment]
same here.

**Comment by restyler** [comment]
This is a problem related to new lines inside the key. I managed to fix the error by switching the key format input type to "expression"

![2024-08-25 at 13 27](https://github.com/user-attachments/assets/4bed527b-b4b2-4331-b5f2-1bd85b81a596)

**Comment by surenka11** [comment]
> This is a problem related to new lines inside the key. I managed to fix the error by switching the key format input type to "expression"
> 
> ![2024-08-25 at 13 27](https://private-user-images.githubusercontent.com/775507/361198927-4bed527b-b4b2-4331-b5f2-1bd85b81a596.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5OTU5MjgsIm5iZiI6MTcyNDk5NTYyOCwicGF0aCI6Ii83NzU1MDcvMzYxMTk4OTI3LTRiZWQ1MjdiLWI0YjItNDMzMS1iNWYyLTFiZDg1YjgxYTU5Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgzMFQwNTI3MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OWFhMmNmZjA5YzM1MGIxMTYxMDAyNzY0NDkwYzg1ZDk1OWI4YjBlZjY2NTI5ZTZhZDExYmRiNWViYzEzMjhmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6VZU0qtBeWCZZULc4HrcKerXng30lgpt7C-iLtTOrIA)

Dear @restyler,

Thank you for your suggestion. Unfortunately, it's not working on my side. Could you please share the ending of the key where you put "}}"? In my case, I have it at the end as: -----END PRIVATE KEY-----\n" }}.

**Comment by restyler** [comment]
> > This is a problem related to new lines inside the key. I managed to fix the error by switching the key format input type to "expression"
> > ![2024-08-25 at 13 27](https://private-user-images.githubusercontent.com/775507/361198927-4bed527b-b4b2-4331-b5f2-1bd85b81a596.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5OTU5MjgsIm5iZiI6MTcyNDk5NTYyOCwicGF0aCI6Ii83NzU1MDcvMzYxMTk4OTI3LTRiZWQ1MjdiLWI0YjItNDMzMS1iNWYyLTFiZDg1YjgxYTU5Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgzMFQwNTI3MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OWFhMmNmZjA5YzM1MGIxMTYxMDAyNzY0NDkwYzg1ZDk1OWI4YjBlZjY2NTI5ZTZhZDExYmRiNWViYzEzMjhmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6VZU0qtBeWCZZULc4HrcKerXng30lgpt7C-iLtTOrIA)
> 
> Dear @restyler,
> 
> Thank you for your suggestion. Unfortunately, it's not working on my side. Could you please share the ending of the key where you put "}}"? In my case, I have it at the end as: -----END PRIVATE KEY-----\n" }}.

![2024-08-30 at 19 52](https://github.com/user-attachments/assets/48af359f-aed2-47da-adf8-2d1e97b29aaf)

**Comment by FredMichel** [comment]
Same here

**Comment by davivc** [comment]
Had the same problem here using the Vertex node and the solution proposed by @restyler worked for me. It isn't the ideal because the key remains exposed.

**Comment by surenka11** [comment]
@davivc Thanks for your comment. I have a question. I managed to connect successfully. But I can not load any project. Did you have the same issue? Thanks a lot.
![1](https://github.com/user-attachments/assets/e500022f-4bbf-454b-9652-a7a85ad4c52f)
![2](https://github.com/user-attachments/assets/c176585c-8823-4728-b935-cf5220a23f62)

**Comment by davivc** [comment]
@surenka11 Yes, it didn't load the projects for me as well. I used the option By ID and wrote manually then it worked. It's painful to do this that way, the credential is not secure but for prototyping now it's fine.

**Comment by hanamizuki** [comment]
I'm having the same issue and @restyler's solution (using expression on private key) doesn't work for me. Do you have other ideas?

Error message
```
Error: error:1E08010C:DECODER routines::unsupported at Sign.sign (node:internal/crypto/sig:128:29) at Object.sign (/usr/local/lib/node_modules/n8n/node_modules/jwa/index.js:152:45) at Object.jwsSign [as sign] (/usr/local/lib/node_modules/n8n/node_modules/jws/lib/sign-stream.js:32:24) at GoogleToken.requestToken (/usr/local/lib/node_modules/n8n/node_modules/gtoken/build/src/index.js:232:31) at GoogleToken.getTokenAsyncInner (/usr/local/lib/node_modules/n8n/node_modules/gtoken/build/src/index.js:166:21) at GoogleToken.getTokenAsync (/usr/local/lib/node_modules/n8n/node_modules/gtoken/build/src/index.js:145:55) at GoogleToken.getToken (/usr/local/lib/node_modules/n8n/node_modules/gtoken/build/src/index.js:97:21) at JWT.refreshTokenNoCache (/usr/local/lib/node_modules/n8n/node_modules/google-auth-library/build/src/auth/jwtclient.js:172:36) at JWT.refreshToken (/usr/local/lib/node_modules/n8n/node_modules/google-auth-library/build/src/auth/oauth2client.js:153:24) at JWT.getRequestMetadataAsync (/usr/local/lib/node_modules/n8n/node_modules/google-auth-library/build/src/auth/oauth2client.js:298:28)
```

**Comment by janober** [comment]
Fix got released with [`n8n@1.60.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.60.0)
